Source/Hydro/Godunov/Godunov_utils_2D.H

```diff
@@ -85,109 +91,141 @@ CAMR_transd (
     const int n = pmap.upassMap[ipass];
     const int nqp = pmap.qpassMap[ipass];
     const amrex::Real srcpass = srcQ(iv, nqp);
-    const amrex::Real compn = no_cut_face ? cdtdx * (flxx(ivp0, n) - flxx(iv, n)) : 0.0;
-    amrex::Real rr = qyp(iv, QRHO);
-    amrex::Real rrnew = rr - flxrho;
-    amrex::Real compo = rr * qyp(iv, nqp) - compn;
-    qp(iv, nqp) = compo / rrnew + hdt * srcpass;
-
-    rr = qym(ivp1, QRHO);
-    rrnew = rr - flxrho;
-    compo = rr * qym(ivp1, nqp) - compn;
-    qm(ivp1, nqp) = compo / rrnew + hdt * srcpass;
+    const amrex::Real compn = no_cov_face ? cdtdx * (flxx(ivpt, n) - flxx(iv, n)) : 0.0;
+
+    if (norm_area(iv) > amrex::Real(0.0)) {
+        amrex::Real rr = qnormp(iv, QRHO);
+        amrex::Real rrnew = rr - flxrho;
+        amrex::Real compo = rr * qnormp(iv, nqp) - compn;
+        qp(iv, nqp) = compo / rrnew + hdt * srcpass;
+    } else {
+        qp(iv, nqp) = qnormp(iv, nqp);
+    }
+
+    if (norm_area(ivpn) > amrex::Real(0.0)) {
+        amrex::Real rr = qnormm(ivpn, QRHO);
+        amrex::Real rrnew = rr - flxrho;
+        amrex::Real compo = rr * qnormm(ivpn, nqp) - compn;
+        qm(ivpn, nqp) = compo / rrnew + hdt * srcpass;
+    } else {
+        qm(iv, nqp) = qnormm(iv, nqp);
```

Comment:
I think this should be `qm(ivpn, nqp) = qnormm(ivpn, nqp);` ?

Reply:
yup!